test/integration/java_purl_test.go

```diff
@@ -0,0 +1,219 @@
+package integration
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/anchore/syft/syft"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger"
+	"github.com/anchore/syft/syft/sbom"
+	"github.com/anchore/syft/syft/source"
+)
+
+func TestJavaPURLs(t *testing.T) {
+	testImage := "anchore/test_images:java-56d52bc"
+	sbom := getCatalog(t, testImage)
+	found := make(map[string]string)
+	for _, p := range sbom.Artifacts.Packages.Sorted() {
+		if p.Type != pkg.JavaPkg && p.Type != pkg.JenkinsPluginPkg {
+			continue
+		}
+		key := fmt.Sprintf("%s@%s", p.Name, p.Version)
+		found[key] = p.PURL
+	}
+	for key, expectedPURL := range expectedPURLs {
+		purl := found[key]
+		assert.Equal(t, expectedPURL, purl, fmt.Sprintf("found wrong or missing PURL for %s want %s, got %s", key, expectedPURL, purl))
+	}
+	for key, foundPURL := range found {
+		expectedPURL := expectedPURLs[key]
+		assert.Equal(t, expectedPURL, foundPURL, fmt.Sprintf("found extra purl for %s want %s, got %s", key, expectedPURL, foundPURL))
+	}
+}
+
+func getCatalog(t *testing.T, image string) sbom.SBOM {
+	detection, err := source.Detect(image, source.DefaultDetectConfig())
+	require.NoError(t, err)
+	theSource, err := detection.NewSource(source.DefaultDetectionSourceConfig())
+	packages, relationships, distro, error := syft.CatalogPackages(theSource, cataloger.DefaultConfig())
+	require.NoError(t, error)
+	sbom := sbom.SBOM{
+		Artifacts: sbom.Artifacts{
+			Packages:          packages,
+			LinuxDistribution: distro,
+		},
+		Relationships: relationships,
+		Source:        theSource.Describe(),
+	}
+	return sbom
+}
+
+// Constructed by:
+// syft anchore/test_images:java-56d52bc -o template -t /tmp/test.templ | grep 'pkg:maven' | sort | uniq >> test/integration/java_purl_test.go
+// where the template is:
+/*
+{{ range .Artifacts}}"{{.Name}}@{{.Version}}":"{{.PURL}}",
+{{ end }}
+*/
+var expectedPURLs = map[string]string{
```

Comment:
TODO: we should hand-check these for other errors.

Reply:
want some help with this in a divide-and-conquer fashion?